tests/raftstore/test_split_region.rs

```diff
@@ -192,3 +192,48 @@ fn test_server_auto_split_region() {
     let mut cluster = new_server_cluster(0, count);
     test_auto_split_region(&mut cluster);
 }
+
+fn test_delay_split_region<T: Simulator>(cluster: &mut Cluster<T>, count: usize) {
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    let pd_client = cluster.pd_client.clone();
+    let cluster_id = cluster.id();
+
+    let region = pd_client.rl().get_region(cluster_id, b"").unwrap();
+
+    let index = (count - 1) as u64;
+
+    let a1 = bytes::encode_bytes(b"a1");
+    cluster.put(&a1, b"v1");
+
+    let a3 = bytes::encode_bytes(b"a3");
+    cluster.put(&a3, b"v3");
+
+    cluster.stop_node(index);
+
+    let a2 = bytes::encode_bytes(b"a2");
+    cluster.split_region(region.get_id(), Some(a2.clone()));
+
+    cluster.run_node(index);
+
+    cluster.put(&a3, b"vv3");
```

Comment:
yes, if election timeout, it will restart a vote, or the leader heartbeat comes. 
But at this time, the new region leader will already send the heartbeat to this node too.